itest-flake: invoice routing hints channel open timeout

```
=== RUN   TestLightningNetworkDaemon/48-of-82/btcd/revoked_uncooperative_close_retribution_remote_hodl
    test_harness.go:88: Failed: (invoice routing hints): exited with error: 
        *errors.errorString timed out waiting for channel open alice-bob: channel not opened before timeout
        /home/travis/gopath/src/github.com/lightningnetwork/lnd/lntest/itest/lnd_routing_test.go:1312 (0xf27fcf)
        	testInvoiceRoutingHints: t.Fatalf("timed out waiting for channel open %s: %v",
        /home/travis/gopath/src/github.com/lightningnetwork/lnd/lntest/itest/test_harness.go:112 (0xeeb1ee)
        	(*harnessTest).RunTestCase: testCase.test(h.lndHarness, h)
        /home/travis/gopath/src/github.com/lightningnetwork/lnd/lntest/itest/lnd_test.go:11702 (0xf9c7af)
        	TestLightningNetworkDaemon.func4: ht.RunTestCase(testCase)
        /home/travis/.gimme/versions/go1.16.3.linux.amd64/src/testing/testing.go:1193 (0x51c72f)
        	tRunner: fn(t)
        /home/travis/.gimme/versions/go1.16.3.linux.amd64/src/runtime/asm_amd64.s:1371 (0x471c01)
        	goexit: BYTE	$0x90	// NOP
=== RUN   TestLightningNetworkDaemon/71-of-82/btcd/sendpayment_amp_invoice/native_payaddr
=== CONT  TestLightningNetworkDaemon
    lnd_test.go:11710: Failure time: 2021-07-01 23:50:50.190
--- FAIL: TestLightningNetworkDaemon (198.52s)
    --- PASS: TestLightningNetworkDaemon/21-of-82/btcd/multi-hop_payments (33.77s)
    --- PASS: TestLightningNetworkDaemon/22-of-82/btcd/single-hop_send_to_route (51.85s)
        --- PASS: TestLightningNetworkDaemon/22-of-82/btcd/single-hop_send_to_route/regular_main_sync (8.18s)
        --- PASS: TestLightningNetworkDaemon/22-of-82/btcd/single-hop_send_to_route/regular_main_stream (8.00s)
        --- PASS: TestLightningNetworkDaemon/22-of-82/btcd/single-hop_send_to_route/regular_routerrpc_sync (8.06s)
        --- PASS: TestLightningNetworkDaemon/22-of-82/btcd/single-hop_send_to_route/mpp_main_sync (7.99s)
        --- PASS: TestLightningNetworkDaemon/22-of-82/btcd/single-hop_send_to_route/mpp_main_stream (8.12s)
        --- PASS: TestLightningNetworkDaemon/22-of-82/btcd/single-hop_send_to_route/mpp_routerrpc_sync (7.99s)
    --- PASS: TestLightningNetworkDaemon/23-of-82/btcd/multi-hop_send_to_route (11.55s)
    --- PASS: TestLightningNetworkDaemon/24-of-82/btcd/send_to_route_error_propagation (12.25s)
    --- PASS: TestLightningNetworkDaemon/25-of-82/btcd/unannounced_channels (4.69s)
    --- PASS: TestLightningNetworkDaemon/26-of-82/btcd/private_channels (17.73s)
    --- PASS: TestLightningNetworkDaemon/27-of-82/btcd/private_channel_update_policy (10.02s)
    --- FAIL: TestLightningNetworkDaemon/28-of-82/btcd/invoice_routing_hints (44.44s)
```

--- a/lntest/itest/lnd_mpp_test.go
+++ b/lntest/itest/lnd_mpp_test.go
@@ -346,9 +346,9 @@ func (c *mppTestContext) waitForChannels() {
 			ctxt, _ := context.WithTimeout(ctxb, defaultTimeout)
 			err = node.WaitForNetworkChannelOpen(ctxt, chanPoint)
 			if err != nil {
-				c.t.Fatalf("(%d): timeout waiting for "+
+				c.t.Fatalf("(%v:%d): timeout waiting for "+
 					"channel(%s) open: %v",
-					node.NodeID, point, err)
+					node.Cfg.Name, node.NodeID, point, err)
 			}
 		}
 	}

--- a/lntest/node.go
+++ b/lntest/node.go
@@ -1210,6 +1210,27 @@ func getChanPointFundingTxid(chanPoint *lnrpc.ChannelPoint) ([]byte, error) {
 	return txid, nil
 }
 
+func checkChanPointInGraph(ctx context.Context,
+	node *HarnessNode, chanPoint wire.OutPoint) bool {
+
+	ctxt, cancel := context.WithTimeout(ctx, DefaultTimeout)
+	defer cancel()
+	chanGraph, err := node.DescribeGraph(ctxt, &lnrpc.ChannelGraphRequest{})
+	if err != nil {
+		return false
+	}
+
+	targetChanPoint := chanPoint.String()
+	for _, chanEdge := range chanGraph.Edges {
+		candidateChanPoint := chanEdge.ChanPoint
+		if targetChanPoint == candidateChanPoint {
+			return true
+		}
+	}
+
+	return false
+}
+
 // lightningNetworkWatcher is a goroutine which is able to dispatch
 // notifications once it has been observed that a target channel has been
 // closed or opened within the network. In order to dispatch these
@@ -1318,6 +1339,21 @@ func (hn *HarnessNode) lightningNetworkWatcher(subscribed chan error) {
 				// dispatched if the number of edges seen for
 				// the channel is at least two.
 				if numEdges := hn.openChans[targetChan]; numEdges >= 2 {
+					close(watchRequest.eventChan)
+					continue
+				}
+
+				// Before we add the channel to our set of open
+				// clients, we'll check to see if the channel
+				// is already in the channel graph of the
+				// target node. This lets us handle the case
+				// where a node has already seen a channel
+				// before a notification has been requested,
+				// causing us to miss it.
+				chanFound := checkChanPointInGraph(
+					context.Background(), hn, targetChan,
+				)
+				if chanFound {
 					close(watchRequest.eventChan)
 					continue
 				}

